target_python_sdk/tests/test_init.py

```diff
@@ -387,3 +387,29 @@ def test_get_offers_prefetch(self):
         for response_token in response_tokens:
             self.assertIsNotNone(response_token.get('activity.id'))
             self.assertIsNotNone(response_token.get('experience.id'))
+
+    @responses.activate
+    def test_get_offers_with_trace(self):
+        setup_mock('debug_trace', responses)
+        opts = deepcopy(self.get_offers_options)
+        opts['request']['trace'] = {
+            'authorization_token': 'token',
+            'usage': {
+                'a': 'b',
+                'c': 'd'
+            }
+        }
+        opts['request'] = create_delivery_request(opts['request'])
+        result = self.client.get_offers(opts)
+        self.assertEqual(len(responses.calls), 1)
+        validate_response(self, result)
+
+        if not result.get('response').execute.mboxes:
```

Comment:
I'm not sure this is the most useful test.  You are validating trace elements on the mock that was provided.  SO it's not really validating SDK functionality is it?

Instead of validating the mocked response, you could validate the request that the SDK makes based on your `get_offers` call. For example, when you make this specific `get_offers` call, you pass in a trace object with `authorization_token`.  But the call to delivery api would contain a trace object with camelCase `authorizationToken`.  You can test that the request has the trace object with `authorizationToken` set to `token` see that the request is correct.

BTW, I'm curious, is there a reason you changed it from camelCase to snake_case?  Was it because python prefers snake case?

Reply:
Updated the test to validate the request is correct.  Keeping the response validation as it does make sure the json response is converted to DeliveryResponse correctly.  The snake_case is coming from the openapi generated objects, since that's the pythonic way.  So the generated code does the conversion between camelCase and snake_case for compatibility between the sdk code and Delivery API.

Reply:
👍 Thanks for adding that to the test.  I am torn on the change to snake_case.  I think it makes sense for the code itself, but I worry about changing the request property names -- for consistency sake.  Not a show stopper for this PR, but something I think we should discuss as a team.  Is it worth the change to be consistent with python naming conventions, or would it be better to keep property names camelCase (specifically with respect to delivery request objects passed into get_offers and others) to maintain consistency across all SDKs?

Reply:
Agreed - it certainly warrants a discussion.  I'll bring it up at standup.